lib/TPP/ToBlockLayoutAndBack.cpp

```diff
@@ -587,17 +601,38 @@ struct PropagateThroughElementWiseOp
     : public OpRewritePattern<linalg::GenericOp> {
   using OpRewritePattern<linalg::GenericOp>::OpRewritePattern;
 
-  // Check sinking preconditions: a) all loops must
-  // be parallel.
-  LogicalResult checkPreconditions(linalg::GenericOp linalgOp) const {
-    if (linalgOp.getNumLoops() != linalgOp.getNumParallelLoops())
+  LogicalResult isElementwise(linalg::GenericOp linalgOp) const {
+    if (!linalg::isElementwise(linalgOp))
       return failure();
     return success();
   }
 
+  // Further restrict to identity or minor identity maps.
+  bool hasMinorIdentityMaps(linalg::GenericOp linalgOp) const {
+    return llvm::all_of(linalgOp.getIndexingMapsArray(),
+                        [](AffineMap map) { return map.isMinorIdentity(); });
+  }
+
+  // Require operands to come from a single `unpack` operation.
+  LogicalResult hasOnlyOnePackedOperand(linalg::GenericOp linalgOp) const {
+    bool oneOf = false;
+    for (OpOperand &operand : linalgOp->getOpOperands()) {
+      linalgx::UnPackOp unpackOp =
+          operand.get().getDefiningOp<linalgx::UnPackOp>();
+      if (unpackOp) {
```

Comment:
I'm guessing there aren't many operands, and this logic while meaningful, is a bit convoluted.

Probably be simpler to just count and `return count != 1`?

Reply:
Done.